responses/__init__.py

```diff
@@ -140,19 +142,32 @@ def get_wrapped(func, responses, registry=None):
     if registry is not None:
         responses._set_registry(registry)
 
+    assert_mock = std_mock.patch.object(
+        target=responses,
+        attribute="assert_all_requests_are_fired",
+        new=assert_all_requests_are_fired,
+    )
+
     if inspect.iscoroutinefunction(func):
         # set asynchronous wrapper if requestor function is asynchronous
         @wraps(func)
         async def wrapper(*args, **kwargs):
-            with responses:
-                return await func(*args, **kwargs)
+
+            with assert_mock:
+                with responses:
```

Comment:
```suggestion
            with assert_mock, responses:
```

Could save a layer of indentation :shrug: